executor/src/lib.rs

```diff
@@ -183,3 +141,21 @@ where
     }
     Ok(restored_consensus_output)
 }
+
+#[async_trait]
+impl<T: ExecutionState + 'static + Send + Sync> ExecutionState for Arc<T> {
+    async fn handle_consensus_transaction(
+        &self,
+        consensus_output: &ConsensusOutput,
+        execution_indices: ExecutionIndices,
+        transaction: Vec<u8>,
+    ) {
+        self.as_ref()
+            .handle_consensus_transaction(consensus_output, execution_indices, transaction)
+            .await
+    }
```

Comment:
@andll out of curiosity why is this necessary? Am I getting it right that is mainly for convenience reasons or am I missing something?

Reply:
Yeah, idea is that some places will use Arc<X> as executor state, and some(perhaps unit tests) might want to pass type without Arc